app/cdap/components/DirectiveInput/Components/AutoComplete/index.tsx

```diff
@@ -0,0 +1,236 @@
+        }
+      );
+    } else {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+  };
+
+  useEffect(() => {
+    getUsage();
+  }, [isDirectiveSelected]);
+
+  useEffect(() => {
+    eventEmitter.on(globalEvents.DIRECTIVEUPLOAD, getUsage);
+    const directiveInput = document.getElementById('directive-input-search');
+    const mousetrap = new Mousetrap(directiveInput);
+    mousetrap.bind('up', handleUpArrow);
+    mousetrap.bind('down', handleDownArrow);
+    mousetrap.bind('enter', handleEnterKey);
+    mousetrap.bind('tab', handleTabKey);
+
+    // returned function will be called on component unmount
+    return () => {
+      mousetrap.unbind('up');
+      mousetrap.unbind('down');
+      mousetrap.unbind('enter');
+      mousetrap.unbind('tab');
+      eventEmitter.off(globalEvents.DIRECTIVEUPLOAD, getUsage);
+    };
+  });
+  const handleUpArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (activeSelectionIndex !== 0) {
+      setActiveSelectionIndex(activeSelectionIndex - 1);
+    }
+  };
+
+  const handleDownArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (activeSelectionIndex !== activeResults.length - 1) {
+      setActiveSelectionIndex(activeSelectionIndex + 1);
+    }
+  };
+
+  const handleEnterKey = () => {
+    if (input.length > 0) {
+      if (activeResults[activeSelectionIndex]) {
+        handleRowClick(activeResults[activeSelectionIndex]);
+      } else {
+        onRowClick({
+          target: { value: `${input}` },
+        });
+      }
+    }
+  };
+
+  const handleTabKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (input.length === 0 || input.split(' ').length !== 1) {
+      return;
+    }
+    event.preventDefault();
+    handleEnterKey();
+  };
+
+  useEffect(() => {
+    if (isDirectiveSelected) {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+    searchMatch(directiveInput);
+    setInput(directiveInput);
+  }, [directiveInput]);
+
+  const searchMatch = (query: string) => {
+    let results = [];
+    const input: string = query;
+    const spaceIndex: number = input.indexOf(' ');
+    if (fuse && input.length > 0) {
+      if (!isDirectiveSelected) {
+        results = fuse
+          .search(input)
+          .slice(0, 3)
+          .filter((row, index) => {
+            if (spaceIndex === -1) {
+              return true;
+            }
+            return row.score === 0 && index === 0;
+          })
+          .map((row) => {
+            row.uniqueId = uuidV4();
+            return row;
+          });
+        reverse(results);
+      } else {
+        results = fuse.search(input.split(':')[1]).map((row) => {
+          row.uniqueId = uuidV4();
+          return row;
+        });
+        reverse(results);
+      }
+    }
+    setActiveResults(results);
+    setInput(query);
```

Comment:
what is this query, rename?

Reply:
It's a search string, renamed it to searchString